### PR TITLE
Improve usage documentation

### LINKED ISSUE #789: joint_trajectory_controller inactive with ursim
Hi:
ROS2 Humble Ubuntu22.04 ur3
with doc https://docs.ros.org/en/ros2_packages/humble/api/ur_robot_driver/usage.html

i am testing ursim and moveit

when i run 
`ros2 launch ur_robot_driver ur_control.launch.py ur_type:=<UR_TYPE> robot_ip:=<IP_OF_THE_ROBOT> launch_rviz:=true
`
and I control the robot using VNC, the robot in RViz also moves. However, it doesn't work when I use MoveIt to plan and execute.

so  i  run
`ros2 control list_controllers`
shows 
`speed_scaling_state_broadcaster[ur_controllers/SpeedScalingStateBroadcaster] active
joint_state_broadcaster[joint_state_broadcaster/JointStateBroadcaster] active
force_torque_sensor_broadcaster[force_torque_sensor_broadcaster/ForceTorqueSensorBroadcaster] active
**joint_trajectory_controller[joint_trajectory_controller/JointTrajectoryController] inactive**
forward_position_controller[position_controllers/JointGroupPositionController] inactive
io_and_status_controller[ur_controllers/GPIOController] active`
then i close all 
run 
`ros2 launch ur_robot_driver ur_control.launch.py ur_type:=ur5e robot_ip:=yyy.yyy.yyy.yyy use_fake_hardware:=true launch_rviz:=true
`
The control through VNC cannot synchronize with RViz, but the planning in MoveIt is consistent with RViz.
the  run
`ros2 control list_controllers`
shows
`speed_scaling_state_broadcaster[ur_controllers/SpeedScalingStateBroadcaster] active
joint_state_broadcaster[joint_state_broadcaster/JointStateBroadcaster] active
force_torque_sensor_broadcaster[force_torque_sensor_broadcaster/ForceTorqueSensorBroadcaster] active
**joint_trajectory_controller[joint_trajectory_controller/JointTrajectoryController] active**
forward_position_controller[position_controllers/JointGroupPositionController] inactive
io_and_status_controller[ur_controllers/GPIOController] active`

I also tried the example 
`ros2 launch ur_robot_driver test_joint_trajectory_controller.launch.py
`
same result 

i also tried 
`Fake hardware on ROS2 Galactic[](https://docs.ros.org/en/ros2_packages/humble/api/ur_robot_driver/usage.html#fake-hardware-on-ros2-galactic)
Currently, the scaled_joint_trajectory_controller does not work on ROS2 Galactic. There is an [upstream Merge-Request](https://github.com/ros-controls/ros2_control/pull/635) pending to fix that. Until this is merged and released, please change the default controller in the [controllers.yaml](https://docs.ros.org/en/ros2_packages/humble/api/ur_robot_driver/ur_moveit_config/config/controllers.yaml) file. Make sure that the default field is assigned true for the joint_trajectory_controller and false for the scaled_joint_trajectory_controller.`
also not work 
what should i do?

thank you!

### LINKED ISSUE #934: Update the ROS node documentation
### Feature summary

Update the [ROS node documentation](https://docs.ros.org/en/ros2_packages/rolling/api/ur_robot_driver/ROS_INTERFACE.html#nodes).
It seems we are missing, at least, the topics the control node is sending out!

The Dashboard server seems more complete as it also have the [advertised-services](https://docs.ros.org/en/ros2_packages/rolling/api/ur_robot_driver/ROS_INTERFACE.html#advertised-services)

In addition, it could be nice if this was autogenerated from the code

### Tasks
*To complete this issue involves*
- [ ] Make documentation

### LINKED ISSUE #1044: Document robot not ready to receive control commands
In the ROS 1 FAQ we had it [written explicitly](https://github.com/UniversalRobots/Universal_Robots_ROS_Driver?tab=readme-ov-file#i-started-everything-but-i-cannot-control-the-robot) what to do if the robot doesn't move.

We should write that down explicitly in the ROS 2 documentation, as well.

### PR DESCRIPTION
**Pull request by fmauch**
closes #934, closes #789 and closes #1044.

This should hopefully significantly improve the usage documentation for end users and clarify a couple of things.

I'm not done, yet, hence the draft status.

### CODE REVIEW

File: ur_robot_driver/doc/usage/startup.rst [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+Finish startup on the robot
+---------------------------
+
+Unless :ref:`headless_mode` is used, you will now have to start the *External Control* program on
+the robot that you have created earlier.
+
+Depending on the :ref:`robot control mode<operation_modes>` do the following:
+
+* In local control mode, load the program on the robot and press the "Play" button |play_button| on the teach pendant.
+* In remote control mode load and start the program using the following dashboard calls:
```
Comment (urrsk) [inline ur_robot_driver/doc/usage/startup.rst:63]:
```suggestion
* In _local control mode_, load the program on the robot and press the "Play" button |play_button| on the teach pendant.
* In _remote control mode_ load and start the program using the following dashboard calls:
```

File: ur_robot_driver/doc/usage/startup.rst [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+* When the driver is started with ``headless_mode:=true`` nothing is needed. The driver is running
+  already.
+
+.. _continuation_after_interruptions:
+
+Continuation after interruptions
+--------------------------------
+
+Whenever the *External Control* program gets interrupted, it has to be unpaused / restarted.
```
Comment (urrsk) [inline ur_robot_driver/doc/usage/startup.rst:78]:
```suggestion
Whenever the *External Control URCap* program gets interrupted, it has to be unpaused / restarted.
```

File: ur_robot_driver/doc/usage/startup.rst [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+* The communication is stopped, since the external source did not receive a command in time.
+* There was another script sent for execution e.g.
+
+  * Script code was sent to the robot via its primary interface
+  * Robot motion is performed using the Teach pendant
+
+Depending on the operation mode, perform one of the following steps:
+
+* In local control mode, simply press the "Play" button |play_button| on the teach pendant.
+* In remote control mode start the program using the following dashboard call:
```
Comment (urrsk) [inline ur_robot_driver/doc/usage/startup.rst:95]:
```suggestion
* In _local control mode_, simply press the "Play" button |play_button| on the teach pendant.
* In _remote control mode_ start the program using the following dashboard call:
```

File: ur_robot_driver/doc/usage/toc.rst [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+interface. Ready to receive control commands.``
+
+.. note::
+
+   When interacting with the teach pendant, or sending other primary programs to the robot, the
+   program will be stopped. On the ROS terminal you will see an output ``Connection to reverse interface dropped.``. Please see the section :ref:`continuation_after_interruptions` for details on how to continue.
+
+In general, make sure you're starting up the robot as explained :ref:`here<ur_robot_driver_startup>`.
+
+When starting the program on the TP, I get an error "The connection to the remote PC could not be established"
```
Comment (urrsk) [inline ur_robot_driver/doc/usage/toc.rst:87]:
```suggestion
When starting the program on the teach pendant, I get an error "The connection to the remote PC could not be established"
```

**Review by urrsk — CHANGES_REQUESTED** [review]

**Comment by urrsk** [comment]
- [ ] Renaming all `ros_control` to `ros2_control`

File: ur_robot_driver/doc/operation_modes.rst [hunk 5]
```diff
@@ -0,0 +1,72 @@
+.. _operation_modes:
+
+Modes of operation
+==================
+
+There are two different sets of **modes** that are used throughout this documentation:
+
+- The driver's mode of operation (External Control vs Headless)
+- (e-series only) The robot's control mode (Local Control Mode vs Remote Control Mode)
```
[the remark this answers is not in the document]
Comment (fmauch) [reply ur_robot_driver/doc/operation_modes.rst:9]: I like that very much: "Teach pendant mode"

Maybe we can should make that change at other places, as well.

**Review by urrsk — APPROVED** [review]

**Pull request merged**